darglint/integrity_checker.py

```diff
@@ -113,24 +116,35 @@ def run_checks(self, function):
                 self._check_yield()
                 self._check_raises()
                 self._sorted = False
-            except ParserException as exception:
-                # If a syntax exception was raised, we may still
-                # want to ignore it, if we place a noqa statement.
-                if (
-                    SYNTAX_NOQA.search(function.docstring)
-                    or EXPLICIT_GLOBAL_NOQA.search(function.docstring)
-                    or BARE_NOQA.search(function.docstring)
-                ):
-                    return
-                if self.raise_errors:
-                    raise
+            # If not docstring is given, and raise on missing is true, and its not
+            # a private function, add an error.
+            elif self.config.raise_on_missing_docstrings and not function.name.startswith("_"):
+                cls = (MissingDocstringForPublicMethod
```

Comment:
This would probably be more understandable if we moved the if-else to the top level, like

```
if function.is_method:
    self.errors.append(
        ...
    )
else:
    self.errors.append(
        ...
    )
```

That will make it more maintainable as well. (Say, if we decide we need to pass more information for functions, or if we change the `__init__` signature for these twe errors.)